File: src/backend/scripts/convert-csv-to-sql.ts
import * as fs from 'fs';
import * as path from 'path';
import * as csv from 'csv-parse/sync';

interface Challenge {
  id: string;
  title: string;
  description: string;
  goal: string;
  steps: string;
  example_dialogue: string;
  tip: string;
  pillar_id: string;
  age_range: string;
  difficulty_level: string;
}

const csvFilePath = path.join(__dirname, '../data/seed_challenges__150_.csv');
const sqlFilePath = path.join(__dirname, '../data/seed_challenges.sql');

// Read CSV file
const fileContent = fs.readFileSync(csvFilePath, 'utf-8');
const records = csv.parse(fileContent, {
  columns: true,
  skip_empty_lines: true
}) as Challenge[];

// Generate SQL statements
const sqlStatements = records.map(record => {
  const escapedTitle = record.title.replace(/'/g, "''");
  const escapedDescription = record.description.replace(/'/g, "''");
  const escapedGoal = record.goal.replace(/'/g, "''");
  const escapedExampleDialogue = record.example_dialogue.replace(/'/g, "''");
  const escapedTip = record.tip.replace(/'/g, "''");

  return `INSERT INTO challenges (
    id, title, description, goal, steps, example_dialogue, tip, pillar_id, age_range, difficulty_level
  ) VALUES (
    '${record.id}',
    '${escapedTitle}',
    '${escapedDescription}',
    '${escapedGoal}',
    '${record.steps}',
    '${escapedExampleDialogue}',
    '${escapedTip}',
    ${record.pillar_id},
    '${record.age_range}',
    ${record.difficulty_level}
  );`;
});

// Write SQL file
fs.writeFileSync(sqlFilePath, sqlStatements.join('\n\n'));
console.log('SQL file generated successfully!'); 

File: src/backend/scripts/convert_csv_to_sql.py
import csv
import json

def escape_sql_string(s):
    if s is None:
        return 'NULL'
    return "'" + str(s).replace("'", "''") + "'"

def convert_csv_to_sql():
    # Read the CSV file
    with open('src/backend/data/seed_challenges__150_.csv', 'r') as csvfile:
        reader = csv.DictReader(csvfile)
        
        # Create SQL file
        with open('src/backend/data/seed_challenges_full.sql', 'w') as sqlfile:
            # Write the DELETE statement
            sqlfile.write('-- Clear existing data\n')
            sqlfile.write('DELETE FROM challenges;\n\n')
            
            # Write the INSERT statement header
            sqlfile.write('-- Insert all 150 challenges\n')
            sqlfile.write('INSERT INTO challenges (id, title, description, goal, steps, example_dialogue, tip, pillar_id, age_range, difficulty_level) VALUES\n')
            
            # Process each row
            rows = list(reader)
            for i, row in enumerate(rows):
                # Convert steps from string to proper JSON
                steps = json.loads(row['steps'])
                
                # Create the VALUES part
                values = [
                    escape_sql_string(row['id']),
                    escape_sql_string(row['title']),
                    escape_sql_string(row['description']),
                    escape_sql_string(row['goal']),
                    escape_sql_string(json.dumps(steps)),
                    escape_sql_string(row['example_dialogue']),
                    escape_sql_string(row['tip']),
                    row['pillar_id'],
                    escape_sql_string(row['age_range']),
                    row['difficulty_level']
                ]
                
                # Write the row
                sqlfile.write('(' + ', '.join(values) + ')')
                
                # Add comma or semicolon
                if i < len(rows) - 1:
                    sqlfile.write(',\n')
                else:
                    sqlfile.write(';\n')

if __name__ == '__main__':
    convert_csv_to_sql() 